[DOCS] Flesh out the user and volunteer documentation

rahulporuri opened the pull request:
## What type of PR is this? (check all applicable)

- [x] 📕Documentation

## Description

This PR fleshes out the user and volunteer documentation. This PR also introduces an additional dependency to the documentation - [mkdocs-glightbox](https://blueswen.github.io/mkdocs-glightbox/) - which enables readers to click on an image to make it larger.

Please see the successful docs build at https://github.com/rahulporuri/fossunited/actions/runs/11408786516/job/31747707132 and see the built docs at https://rahulporuri.github.io/fossunited/.

This PR also addresses an existing issue with a bad logo and favicon path for the docs - see https://fossunited.github.io/fossunited/

## Checklist

- [x] I have read the [contribution guidelines]("./docs/CONTRIBUTING.md").
- [x] I have tested these changes locally.
- [x] I have updated the documentation (If applicable).
- [ ] ~The code follows the project's coding standards.~
- [ ] ~I have added/updated tests, if applicable.~

rahulporuri committed:
DOCS: Flesh out the documentation

please note that this commit introduces an additional dependency on
the mkdocs documentation - mkdocs-glightbox

rahulporuri committed:
DEV: Install additional docs dependency in github action

rahulporuri committed:
FIX: Address trailing whitespace in docs

and format markdown docs

harshtandiya reviewed (approved):
LGTM, great work rahul, Thank you!

rahulporuri committed:
ADD : development and architecture documentation

please note that this commit also updates the mkdocs configuration to
enable parsing and rendering mermaid diagrams to effectively communicate
the architecture of the platform using diagrams

rahulporuri committed:
ADD: placeholders for marketing info and what to do on the actual day of

the event

rahulporuri commented:
apologies for pushing after your review @harshtandiya - could you please take another look at the PR and the updates?

also, I'm not sure why the tests are failing. I'm looking into the linting failure too - precommit isn't happy with the mkdocs yaml file.

rahulporuri committed:
FIX: Allow unsafe yaml extensions

this is needed because of the mermaid extensions needed in the mkdocs
yaml configuration file

rahulporuri commented:
@harshtandiya interesting - i fixed the linter/pre-commit issue and the test failure seems to be intermittent - please take another look when you have the time.

the updated docs can still be seen at https://rahulporuri.github.io/fossunited/ - and once the PR is merged, they will be visible at fossunited.github.io/fossunited - and eventually docs.fossunited.org
